### PR TITLE
Google contacts

### PR DESCRIPTION
**Pull request by olsona**
Integrated Google Contacts - now works end-to-end
Cleaned up testing
Decided, for now, to only upload one name per contact - something to revisit later on

### CODE REVIEW

File: portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java [hunk 1]
```diff
@@ -69,20 +79,46 @@ private static Name convertToGoogleName(StructuredName vCardName) {
...
     if (vCardName.getAltId() == null) {
-      name.setMetadata(PRIMARY_FIELD_METADATA);
+      fieldMetadata.setPrimary(true);
     } else {
-      name.setMetadata(SECONDARY_FIELD_METADATA);
+      fieldMetadata.setPrimary(false);
     }
+
+    String vCardNameSource = vCardName.getParameter(SOURCE_PARAM_NAME_TYPE);
+    if (vCardNameSource == null || vCardNameSource.equals(CONTACT_SOURCE_TYPE)) {
```
Comment (rtannenbaum) [inline portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java]: This seems to contradict the comment above about adding null source. Looks like you are adding null sources here but the comment indicates you're not sure if you can?

File: portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java [hunk 2]
```diff
@@ -69,20 +79,46 @@ private static Name convertToGoogleName(StructuredName vCardName) {
...
 
     return name;
   }
 
+  private static Name getPrimaryGoogleName(List<StructuredName> vCardNameList) {
+    StructuredName primaryVCardName;
+
+    // first look if there's a primary name (or names)
+    // if no primary name exists, simply pick the first "alt" name
+    List<StructuredName> primaryNames = vCardNameList.stream()
```
Comment (rtannenbaum) [inline portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java]: nice streams!

File: portability-core/src/test/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactToVCardConverterTest.java [hunk 3]
```diff
@@ -72,33 +77,37 @@ public void testConversionToVCardEmail() {
...
     assertThat(getValuesFromTextProperties(resultSecondaryEmailList))
         .containsExactly(secondaryString1, secondaryString2);
   }
 
   @Test
   public void testConversionToVCardNames() {
     // Set up Person with a primary name and two secondary names
-    String primaryGivenName = "J. K.";
-    String primaryFamilyName = "Rowling";
+    String primaryGivenName = "Mark";
```
Comment (rtannenbaum) [inline portability-core/src/test/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactToVCardConverterTest.java]: These literary references are over my head, but I still appreciate them 😛

File: portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java (hunk 1 above)
Comment (olsona) [reply portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/VCardToGoogleContactConverter.java]: Whoops, good catch!

File: portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsConstants.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package org.dataportabilityproject.serviceProviders.google.contacts;
+
+public class GoogleContactsConstants {
```
[the remark this answers is not in the document]
Comment (rtannenbaum) [reply portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsConstants.java]: what about this one?

[the remark this answers is not in the document]
Comment (olsona) [reply portability-core/src/main/java/org/dataportabilityproject/serviceProviders/google/contacts/GoogleContactsConstants.java]: Whoops, for some reason that was hidden!

**Pull request merged**